src/main.rs

```diff
@@ -131,82 +223,68 @@ fn main() -> Result<()> {
+                entry.read_to_end(&mut data).context("reading cfpa.bin")?;
                 data
             };
 
-            let img_cmpa: &[u8; 512] = img_cmpa[..].try_into()
+            let img_cmpa: &[u8; 512] = img_cmpa[..]
+                .try_into()
                 .map_err(|_| anyhow!("CMPA file is wrong length!"))?;
-            let img_cfpa: &[u8; 512] = img_cfpa[..].try_into()
+            let img_cfpa: &[u8; 512] = img_cfpa[..]
+                .try_into()
                 .map_err(|_| anyhow!("CFPA file is wrong length!"))?;
 
-            let cmpa = CMPAPage::from_bytes(img_cmpa)
-                .context("parsing CMPA")?;
-            let mut cfpa = CFPAPage::from_bytes(img_cfpa)
-                .context("parsing CFPA")?;
+            let cmpa = CMPAPage::from_bytes(img_cmpa).context("parsing CMPA")?;
+            let mut cfpa = CFPAPage::from_bytes(img_cfpa).context("parsing CFPA")?;
 
             // Basic checks to try and detect mixups.
             log_verify_only_on_failure();
-            lpc55_sign::verify::verify_image(
-                &img_bootleby,
-                cmpa,
-                cfpa.clone(),
-            ).context("verifying image")?;
+            lpc55_sign::verify::verify_image(&img_bootleby, cmpa, cfpa.clone())
+                .context("verifying image")?;
 
             if let Some(keymask) = require_key_enable_shape {
-                // The user has requested that we ensure the CFPA has a
-                // particular set of keys enabled, and no others. Let's get to
-                // it. The ROTKH_REVOKE word is a 32-bit field with the
-                // interesting bits in its least significant byte. In each
-                // field, `0b01` represents enabled, `0b00` represents invalid
-                // (which could become enabled later), and `0b10` represents
-                // revoked.
-                //
-                // We require things to either be enabled or invalid, not
-                // revoked, because that's the use case this was written for.
-                let required = {
-                    let mut bits = 0;
-                    for bit in 0..4 {
-                        if keymask & (1 << bit) != 0 {
-                            bits |= 0b01 << (2 * bit);
-                        }
-                    }
-                    bits
-                };
-                if cfpa.rkth_revoke != required {
+                let found = KeyStateTable::from_rotkh_revoke(cfpa.rkth_revoke);
+                if found != keymask {
                     println!("**** FAILED KEY CHECKS ****");
                     println!("You provided a key shape requirement mask,");
                     println!("but the bundle's CFPA doesn't meet it.");
-                    println!("required: {required:02x}");
-                    println!("found:    {:02x}", cfpa.rkth_revoke);
+                    println!("required: {:?}", keymask);
+                    println!("found:    {:?}", found);
```

Comment:
Style nit, your opinion may vary:
```suggestion
                    println!("required: {keymask:?}");
                    println!("found:    {found:?}");
```

Reply:
(also, considering using `{}` instead of `{:?}`, since we want the `Display` implementation and not `Debug`; this works because there's a default `Debug` implementation that calls into `Display`)